modules/aleph/includes/aleph.reservation.inc

```diff
@@ -258,40 +273,50 @@ function aleph_reservation_create($account, $id) {
   $interest_period = aleph_reservation_default_interest_period();
 
   $end_request_date = new DateTime();
-  $end_request_date->add(new DateInterval('P' . $interest_period . 'D'));
+  try {
+    $end_request_date->add(new DateInterval('P' . $interest_period . 'D'));
+  }
+  catch (Exception $e) {
+  }
 
   // Build the reservation object.
   $defaults = aleph_reservation_default_options($account);
   $reservation = new AlephReservation();
   $request = new AlephRequest();
+  $material = new AlephMaterial();
+  $material->setId($id);
   $request->setRequestDate(new DateTime());
   $request->setEndRequestDate($end_request_date);
   $request->setPickupLocation($defaults['aleph_preferred_branch']);
+  $request->setDocNumber($id);
   $reservation->setRequest($request);
 
+  try {
+    $holding_groups = $patron_handler->getHoldingGroups($patron, $material);
+  }
```

Comment:
This way we retrieve the holding groups from the patron instead of basing them on the configuration on the site. Is that intentional?

Reply:
The holding groups are different from patron to patron.
It is intentional. :)

Reply:
OK. I just worried if we were going to be creating reservations for items outside the scope of the site.